rskj-core/src/main/java/co/rsk/mine/gas/MinGasPriceProviderType.java

```diff
@@ -0,0 +1,7 @@
+package co.rsk.mine.gas;
+
+public enum MinGasPriceProviderType {
+    DEFAULT,
+    HTTP_GET,
+    ETH_CALL
```

Comment:
I have a suggestion to use the enum to our favor and mapping directly according the method configured. We could even set to always return `DEFAULT` in case is anything different from the other two types.

My suggestion is the following:
```suggestion
    DEFAULT("DEFAULT"),
    HTTP_GET("HTTP_GET"),
    ETH_CALL("ETH_CALL");

    private String type;

    MinGasPriceProviderType(final String type){
        this.type = type;
    }

    public static MinGasPriceProviderType fromMethodType(String type) {
        return Arrays.stream(MinGasPriceProviderType.values())
                .filter( mgp -> mgp.type.equals(type))
                .findFirst()
                .orElse(MinGasPriceProviderType.DEFAULT);
    }
```

You can rename anything if you approve the suggestion and apply it.

I have created a test for it and it works perfectly:

```
class MinGasPriceProviderTypeTest {

    @Test
    void fromMethodTypeMapping() {
        //given
        String httpGet = "HTTP_GET";
        String ethCall = "ETH_CALL";
        String defaultType = "DEFAULT";
        String anything = "anything";
        String nullType = null;
        //when
        MinGasPriceProviderType httpGetType = MinGasPriceProviderType.fromMethodType(httpGet);
        MinGasPriceProviderType ethCallType = MinGasPriceProviderType.fromMethodType(ethCall);
        MinGasPriceProviderType defaultTypeType = MinGasPriceProviderType.fromMethodType(defaultType);
        MinGasPriceProviderType anythingType = MinGasPriceProviderType.fromMethodType(anything);
        MinGasPriceProviderType nullTypeType = MinGasPriceProviderType.fromMethodType(nullType);
        //then
        assertEquals(MinGasPriceProviderType.HTTP_GET, httpGetType);
        assertEquals(MinGasPriceProviderType.ETH_CALL, ethCallType);
        assertEquals(MinGasPriceProviderType.DEFAULT, defaultTypeType);
        assertEquals(MinGasPriceProviderType.DEFAULT, anythingType);
        assertEquals(MinGasPriceProviderType.DEFAULT, nullTypeType);
    }
}
```

Reply:
I like your suggestion to work with enums but for this case it is preferred to not return default automatically. As it is now if an user adds a non valid method or wrong naming an exception will be thrown by the config class automatically and the user will be forced to fix it accordingly.

Reply:
Then, just throw an exception on the method instead return the DEFAULT. The enum implementation is unrelated with the default.